### PR TITLE
IGNITE-22987 Add logs for rocksdb flusher

### PR DESCRIPTION
**Pull request by ibessonov**
https://issues.apache.org/jira/browse/IGNITE-22987
Thank you for submitting the pull request.

To streamline the review process of the patch and ensure better code quality
we ask both an author and a reviewer to verify the following:

### The Review Checklist
- [ ] **Formal criteria:** TC status, codestyle, mandatory documentation. Also make sure to complete the following:  
\- There is a single JIRA ticket related to the pull request.  
\- The web-link to the pull request is attached to the JIRA ticket.  
\- The JIRA ticket has the Patch Available state.  
\- The description of the JIRA ticket explains WHAT was made, WHY and HOW.  
\- The pull request title is treated as the final commit message. The following pattern must be used: IGNITE-XXXX Change summary where XXXX - number of JIRA issue.
- [ ] **Design:** new code conforms with the design principles of the components it is added to.
- [ ] **Patch quality:** patch cannot be split into smaller pieces, its size must be reasonable.
- [ ] **Code quality:** code is clean and readable, necessary developer documentation is added if needed.
- [ ] **Tests code quality:** test set covers positive/negative scenarios, happy/edge cases. Tests are effective in terms of execution time and resources.

### Notes
- [Apache Ignite Coding Guidelines](https://cwiki.apache.org/confluence/display/IGNITE/Java+Code+Style+Guide)

### CODE REVIEW

File: modules/raft/src/main/java/org/apache/ignite/internal/raft/util/SharedLogStorageFactoryUtils.java [hunk 1]
```diff
@@ -33,10 +34,15 @@ public class SharedLogStorageFactoryUtils {
     public static final String LOGIT_STORAGE_ENABLED_PROPERTY = "LOGIT_STORAGE_ENABLED";
 
     /** Creates a LogStorageFactory with the {@link DefaultLogStorageFactory} implementation. */
+    @TestOnly
     public static LogStorageFactory create(String nodeName, Path lazyLogStoragePath) {
```
Comment (rpuch) [inline modules/raft/src/main/java/org/apache/ignite/internal/raft/util/SharedLogStorageFactoryUtils.java:38]: Could you please also rename that `lazyLogStoragePath` to `logStoragePath`? It seems that 'lazy' is not relevant anymore

File: modules/storage-rocksdb/src/test/java/org/apache/ignite/internal/storage/rocksdb/RocksDbMvPartitionStorageTest.java [hunk 2]
```diff
@@ -49,7 +49,7 @@ void setUp(
             @WorkDirectory Path workDir,
             @InjectConfiguration("mock.flushDelayMillis = 0")
             RocksDbStorageEngineConfiguration engineConfig,
-            @InjectConfiguration("mock.profiles.default = {engine = rocksdb, size = 16777216, writeBufferSize = 16777216}")
+            @InjectConfiguration("mock.profiles.default = {engine = rocksdb, size = 16777216, writeBufferSize = 67108864}")
```
Comment (rpuch) [inline modules/storage-rocksdb/src/test/java/org/apache/ignite/internal/storage/rocksdb/RocksDbMvPartitionStorageTest.java:52]: I'm curious: why was this change needed?

File: modules/raft/src/main/java/org/apache/ignite/internal/raft/storage/impl/DefaultLogStorageFactory.java [hunk 3]
```diff
@@ -91,6 +93,9 @@ public class DefaultLogStorageFactory implements LogStorageFactory {
     @SuppressWarnings("ThreadLocalNotStaticFinal")
     private final ThreadLocal<WriteBatch> threadLocalWriteBatch = new ThreadLocal<>();
 
+    /** Rocksdb log flusher, without external access. Used for logs only. */
+    private RocksDbFlusher flusher;
```
Comment (rpuch) [inline modules/raft/src/main/java/org/apache/ignite/internal/raft/storage/impl/DefaultLogStorageFactory.java:97]: This looks a bit like an abuse. Is it possible to use a listener instead?

Comment (ibessonov) [reply modules/raft/src/main/java/org/apache/ignite/internal/raft/storage/impl/DefaultLogStorageFactory.java:97]: Sure, but I'd have to refactor the listener a little bit

File: modules/storage-rocksdb/src/test/java/org/apache/ignite/internal/storage/rocksdb/RocksDbMvPartitionStorageTest.java (hunk 2 above)
Comment (ibessonov) [reply modules/storage-rocksdb/src/test/java/org/apache/ignite/internal/storage/rocksdb/RocksDbMvPartitionStorageTest.java:52]: Small buffer size triggers flush on every write, I reduce log output here

File: modules/raft/src/main/java/org/apache/ignite/internal/raft/util/SharedLogStorageFactoryUtils.java (hunk 1 above)
Comment (ibessonov) [reply modules/raft/src/main/java/org/apache/ignite/internal/raft/util/SharedLogStorageFactoryUtils.java:38]: Sure

File: modules/runner/src/main/java/org/apache/ignite/internal/app/IgniteImpl.java [hunk 4]
```diff
@@ -533,7 +533,7 @@ public class IgniteImpl implements Ignite {
         ComponentWorkingDir partitionsWorkDir = partitionsPath(systemConfiguration, workDir);
 
         partitionsLogStorageFactory =
-                SharedLogStorageFactoryUtils.create("data log", clusterSvc.nodeName(), partitionsWorkDir.raftLogPath());
+                SharedLogStorageFactoryUtils.create("table data log", clusterSvc.nodeName(), partitionsWorkDir.raftLogPath());
```
Comment (rpuch) [inline modules/runner/src/main/java/org/apache/ignite/internal/app/IgniteImpl.java:536]: table or partition? We usually use the latter, but IDK which is better here

Comment (ibessonov) [reply modules/runner/src/main/java/org/apache/ignite/internal/app/IgniteImpl.java:536]: Well... depends on a level of abstraction that you choose. That's for us anyway
